gpusim.cpp

```diff
@@ -144,6 +147,10 @@ void GPUSimServer::extractData(const QString& database_fname,
         throw std::runtime_error("Database version incompatible with this GPUSim version");
     }
 
+    char* dbkey_char;
+    datastream >> dbkey_char;
+    dbkey = dbkey_char;
+    delete[] dbkey_char;
```

Comment:
Can't you deserialize right into the QString? Or are you not serializing as a QString?

Reply:
Unfortunately PyQt serializes as Char*'s now with QDataStream.writeString, and if you try to read a QString, Qt doesn't do the right thing, it just tries to read the wrong data out of the stream and it screws everything up.  This was...annoying to figure out.